src/main/java/core/basesyntax/strategy/ReportGenerateImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import java.util.Map;
+
+public class ReportGenerateImpl implements ReportGenerate {
+    public String getReport() {
+        StringBuilder report = new StringBuilder();
+        report.append("fruit,quantity").append(System.lineSeparator());
+        for (Map.Entry<String, Integer> entry : Storage.getAllFruits().entrySet()) {
```

Comment:
Direct access to the Storage class's static method getAllFruits() can make testing difficult and violates encapsulation. Consider introducing a service layer that interacts with the Storage and can be mocked during testing.